Encounter Me.Api/Trail generation Utilities/GooglePoints.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Text;
+
+namespace Encounter_Me.Api.Trail_generation_Utilities
+{
+    /// <summary>
+    /// See https://developers.google.com/maps/documentation/utilities/polylinealgorithm
+    /// </summary>
+    public static class GooglePoints
+    {
+        /// <summary>
+        /// Decode google style polyline coordinates.
+        /// </summary>
+        /// <param name="encodedPoints"></param>
+        /// <returns></returns>
+        public static IEnumerable<Coordinate> Decode(string encodedPoints)
+        {
+            if (string.IsNullOrEmpty(encodedPoints))
+                throw new ArgumentNullException("encodedPoints");
+
+            char[] polylineChars = encodedPoints.ToCharArray();
+            int index = 0;
+
+            int currentLat = 0;
+            int currentLng = 0;
+            int next5bits;
+            int sum;
+            int shifter;
+
+            while (index < polylineChars.Length)
+            {
+                // calculate next latitude
+                sum = 0;
+                shifter = 0;
+                do
+                {
+                    next5bits = (int)polylineChars[index++] - 63;
+                    sum |= (next5bits & 31) << shifter;
+                    shifter += 5;
+                } while (next5bits >= 32 && index < polylineChars.Length);
+
+                if (index >= polylineChars.Length)
+                    break;
+
+                currentLat += (sum & 1) == 1 ? ~(sum >> 1) : (sum >> 1);
+
+                //calculate next longitude
+                sum = 0;
+                shifter = 0;
+                do
+                {
+                    next5bits = (int)polylineChars[index++] - 63;
+                    sum |= (next5bits & 31) << shifter;
+                    shifter += 5;
+                } while (next5bits >= 32 && index < polylineChars.Length);
+
+                if (index >= polylineChars.Length && next5bits >= 32)
+                    break;
```

Comment:
Also, code seems to be duplicated - maybe the duplicating part could be a method of the `Coordinate` class?

Reply:
I agree, we should imply on single responsibility rule more.